src/datasets/exceptions.py

```diff
@@ -5,10 +5,23 @@
 class DatasetsError(Exception):
     """Base class for exceptions in this library."""
 
-    pass
-
 
 class DefunctDatasetError(DatasetsError):
     """The dataset has been defunct."""
 
-    pass
+
+class FileNotFoundDatasetsError(DatasetsError, FileNotFoundError):
```

Comment:
why do you create this intermediate exception? it's not used as far as I can understand

Reply:
Both `DataFilesNotFoundError` and `DatasetNotFoundError` derive from `FileNotFoundDatasetsError`. Therefore both are subclasses of `DatasetsError` (the base error class for all errors raised by `datasets`) and also subclasses of `FileNotFoundError` (so that they are backward compatible).

Reply:
> so that they are backward compatible

OK! I didn't get this part